src/block_ffm.rs

```diff
@@ -412,8 +516,14 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
                 let mut f1_offset_ffmk = f1_offset + f1_ffmk;
 
                 // Self-interaction
-                let mut v = 0.0;
-                for k in f1_offset_ffmk..f1_offset_ffmk + ffmk_as_usize {
+                let mut v_simd = f32x4::splat(0.0);
+                for k in (f1_offset_ffmk..f1_offset_ffmk + ffmk_end).step_by(STEP) {
+                    let contra_fields_simd = f32x4::from_slice(contra_fields.get_unchecked(k .. k + STEP));
+                    v_simd += contra_fields_simd * contra_fields_simd;
+                }
+
+                let mut v = v_simd.reduce_sum();
```

Comment:
Is there some kind of equality guarantee for f32 and https://doc.rust-lang.org/nightly/core/simd/trait.SimdFloat.html? If not, are there any provable/apparent diffs we should be aware of?

Reply:
aggregations could be subject to varying numerics otherwise, worth checking maybe

Reply:
I am creating separate models to test the weights & predictions for any inconsistencies.